index.d.ts

```diff
@@ -1,54 +1,52 @@
-import * as mongodb from 'mongodb';
-import fastify, { FastifyPlugin } from 'fastify';
+import type { FastifyPluginCallback } from 'fastify';
+import type { Db, MongoClient, ObjectId, MongoClientOptions } from 'mongodb';
 
-declare namespace fastifyMongodb {
-  interface FastifyMongoObject {
-    /**
-     * Mongo client instance
-     */
-    client: mongodb.MongoClient;
-    /**
-     * DB instance
-     */
-    db?: mongodb.Db;
-    /**
-     * Mongo ObjectId class
-     */
-    ObjectId: typeof mongodb.ObjectId;
-  }
+export interface FastifyMongoObject {
+  /**
+   * Mongo client instance
+   */
+  client: MongoClient;
+  /**
+   * DB instance
+   */
+  db?: Db;
+  /**
+   * Mongo ObjectId class
+   */
+  ObjectId: typeof ObjectId;
+}
 
-  interface FastifyMongoNestedObject {
-    [name: string]: FastifyMongoObject;
-  }
+export interface FastifyMongoNestedObject {
+  [name: string]: FastifyMongoObject;
+}
 
-  interface FastifyMongodbOptions {
-    /**
-     * Force to close the mongodb connection when app stopped
-     * @default false
-     */
-    forceClose?: boolean;
-    /**
-     * Database name to connect
-     */
-    database?: string;
-    name?: string;
-    /**
-     * Pre-configured instance of MongoClient
-     */
-    client?: mongodb.MongoClient;
-    /**
-     * Connection url
-     */
-    url?: string;
-  }
+export interface FastifyMongodbOptions extends MongoClientOptions {
+  /**
+   * Force to close the mongodb connection when app stopped
+   * @default false
+   */
+  forceClose?: boolean;
+  /**
+   * Database name to connect
+   */
+  database?: string;
+  name?: string;
+  /**
+   * Pre-configured instance of MongoClient
+   */
+  client?: MongoClient;
+  /**
+   * Connection url
+   */
+  url?: string;
 }
 
 declare module 'fastify' {
   interface FastifyInstance {
-    mongo: fastifyMongodb.FastifyMongoObject & fastifyMongodb.FastifyMongoNestedObject;
+    mongo: FastifyMongoObject & FastifyMongoNestedObject;
   }
 }
 
-declare const fastifyMongodb: FastifyPlugin<fastifyMongodb.FastifyMongodbOptions>;
+export const fastifyMongodb: FastifyPluginCallback<FastifyMongodbOptions>;
```

Comment:
Why are you exporting twice? It's already exported in `export default`

Reply:
It allow the usage of `import { fastifyMongodb } from 'fastify-mongodb'` and this export is added by `fastify-plugin`.

Reply:
but are we currently doing it in another plugin? looks like a new behavior that is not documented so far.

Reply:
We are doing only the default export or I'm missing something? https://github.com/fastify/fastify-mongodb/blob/master/index.js#L103

Reply:
When we are using `fastify-plugin`, it will use the `plugin name` to set another export.
https://github.com/fastify/fastify-plugin/blob/master/plugin.js#L59

It is old behavior, but `FastifyPluginCallback` cannot reflect this behavior. So, we need to export it in each `plugin` which use `fastify-plugin`.

Reply:
I was unaware of it. Thanks!